lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -237,7 +237,7 @@ defmodule Mix.Tasks.Xref do
   ## Print unreachable
 
   defp print_unreachable(entries) do
-    Enum.each(entries, &IO.write(format_unreachable(&1)))
+    Enum.each(entries, &Mix.shell().info(format_unreachable(&1)))
```

Comment:
You will have to change `format_unreachable` and `format_call` to remove [the trailing new line here](https://github.com/jfornoff/elixir/blob/aaa0eb385a3b7280b4bbfd6629aa2f8ddf2901cc/lib/mix/lib/mix/tasks/xref.ex#L249) since the shell info already adds a new line.